tools/autograd/derivatives.yaml

```diff
@@ -1537,22 +1585,23 @@
 
 - name: hardshrink(Tensor self, Scalar lambd=0.5) -> Tensor
   self: hardshrink_backward(grad, self, lambd)
+  result: auto_element_wise
 
 - name: hardshrink_backward(Tensor grad_out, Tensor self, Scalar lambd) -> Tensor
   grad_out: hardshrink_backward(grad, self, lambd)
   self: zeros_like(grad)
 
 - name: hardtanh(Tensor self, Scalar min_val=-1, Scalar max_val=1) -> Tensor
-  self: hardtanh_backward(grad, self, min_val, max_val)
-
-- name: hardtanh_(Tensor(a!) self, Scalar min_val=-1, Scalar max_val=1) -> Tensor(a!)
   self: hardtanh_backward(grad, result, min_val, max_val)
+  result: auto_element_wise
 
 - name: leaky_relu(Tensor self, Scalar negative_slope=0.01) -> Tensor
   self: leaky_relu_backward(grad, self, negative_slope, false)
+  result: auto_element_wise
 
 - name: leaky_relu_(Tensor(a!) self, Scalar negative_slope=0.01) -> Tensor(a!)
   self: leaky_relu_backward(grad, result, negative_slope, true)
+  result: leaky_relu_backward(self_t, result, negative_slope, true)
```

Comment:
What goes wrong if this is `auto_element_wise`?

Reply:
Nothing :D Changing.